### PR TITLE
Ferrocene example fixes and Aarch64 Armv8-A Heap Demo

### PR DESCRIPTION
**Pull request by jonathanpallant**
* Fixes to the QEMU Armv8-R demo
* Added a heap to the QEMU Armv8-A demo

### CODE REVIEW

File: example-code/qemu-armv8r/src/critical_section.rs [hunk 1]
```diff
@@ -19,7 +19,6 @@ pub fn interrupts_enabled() -> bool {
 unsafe impl critical_section::Impl for SingleCoreCriticalSection {
     unsafe fn acquire() -> critical_section::RawRestoreState {
         let was_active = interrupts_enabled();
-        // NOTE: Fence guarantees are provided by interrupt::disable(), which performs a `compiler_fence(SeqCst)`.
```
Comment (miguelraz) [inline example-code/qemu-armv8r/src/critical_section.rs:22]: Not blocking, but - Is there a reason this note was removed? It wasn't mentioned in the commit.

Comment (jonathanpallant) [reply example-code/qemu-armv8r/src/critical_section.rs:22]: The comment is wrong (we're no longer calling that function)

Comment (miguelraz) [reply example-code/qemu-armv8r/src/critical_section.rs:22]: That will do it.

File: example-code/qemu-aarch64v8a/README.md [hunk 2]
```diff
@@ -73,37 +88,48 @@ This demo includes a [`build.sh`](./build.sh) shell script to build our binary
...
+Running rustc for with_heap...
+Generating asm for with_heap...
+Generating map for with_heap...
 $ qemu-system-aarch64 \
     -machine virt \
     -cpu cortex-a57 \
     -semihosting \
     -nographic \
-    -kernel ./target/production/basic-rust
+    -kernel target/production/with_heap
```
Comment (miguelraz) [inline example-code/qemu-aarch64v8a/README.md:120]:
```suggestion
$ qemu-system-aarch64 \
    -machine virt \
    -cpu cortex-a57 \
    -semihosting \
    -nographic \
    -kernel target/production/with_heap
```
$ ./qemu.sh

**Pull request merged**